### PR TITLE
Added configuration for doxygen build

### PR DESCRIPTION
**Pull request by tammyleino**
Generate documentation with "doxygen Doxyfile"
Signed-off-by: Tammy Leino <tammy_leino@mentor.com>

### CODE REVIEW

File: doc/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,19 @@
+if (DOXYGEN_FOUND)
```
Comment (arnopo) [inline doc/CMakeLists.txt:1]: With this commit, there will be a `doc` folder and a `docs` folder.
We will have to put everything in the `doc` folder, but that can be done later.

**Review by arnopo — APPROVED** [review]

File: cmake/options.cmake [hunk 2]
```diff
@@ -114,4 +114,6 @@ if (DEFINED RPMSG_BUFFER_SIZE)
   add_definitions( -DRPMSG_BUFFER_SIZE=${RPMSG_BUFFER_SIZE} )
 endif (DEFINED RPMSG_BUFFER_SIZE)
 
+option (WITH_DOC "Build with documentation" ON)
```
Comment (arnopo) [inline cmake/options.cmake:117]: There are a lot of warnings during doc generation. the function and structure headers do not respect the doxygen syntax.
Please deactive the doc generation by default, until it is fixed

**Review by arnopo — CHANGES_REQUESTED** [review]

File: doc/CMakeLists.txt (hunk 1 above)
Comment (tammyleino) [reply doc/CMakeLists.txt:1]: @arnopo I discussed this with @danmilea and his perspective was, "The doc directory mirrors libmetal and it contains rules to build the docs. open-amp/docs on the other hand already contains some docs in mardown format. I wouldn't mix them."

File: cmake/options.cmake (hunk 2 above)
Comment (tammyleino) [reply cmake/options.cmake:117]: @arnopo I will see if I can find a way to disable these warnings.

Comment (tammyleino) [reply cmake/options.cmake:117]: @arnopo I disabled warnings - please let me know if this is suitable.

File: doc/CMakeLists.txt (hunk 1 above)
Comment (tammyleino) [reply doc/CMakeLists.txt:1]: @arnopo I have removed the docs folder and placed everything in the doc folder.

File: cmake/options.cmake (hunk 2 above)
Comment (arnopo) [reply cmake/options.cmake:117]: I still have warnings. Here is the script I'm using
```
#!/bin/bash
 
dirs_to_check=$PWD
echo "################### build libmetal #####################"
cd libmetal
rm -r build
cmake . -Bbuild -DCMAKE_C_FLAGS="-Wall -Wextra -Wshadow -Wunused-but-set-variable" || exit 1
cd build
make || exit 1

cd $dirs_to_check

echo "################### build openamp #####################"
cd open-amp
rm -r build*
cmake . -DCMAKE_C_FLAGS="-Wall -Wextra -Wshadow -Wunused-but-set-variable" -DWITH_APPS=on -DWITH_DCACHE_BUFFERS=on -DWITH_PROXY=on -Bbuild -DCMAKE_INCLUDE_PATH="../libmetal/build/lib/include" -DCMAKE_LIBRARY_PATH="../libmetal/build/lib" || exit 1
cd build
make install  DESTDIR=./bin || exit 1
```
with following project hierarchy:
```
openamp_project 
|_ open-amp
|_ libmetal
```
More than that  I would prefer to not  mask warnings, we could forget to re-enable the option.
By setting` WITH_DOC` to OFF by default this just keep actual state of the doc generation. 
then to force the doc generation we have just to add `-DWITH_DOC=on` option in the cmake command
`cmake . -DCMAKE_C_FLAGS="-Wall -Wextra -Wshadow -Wunused-but-set-variable" -DWITH_APPS=on -DWITH_DCACHE_BUFFERS=on -DWITH_PROXY=on -Bbuild -DWITH_DOC=on -DCMAKE_INCLUDE_PATH="../libmetal/build/lib/include" -DCMAKE_LIBRARY_PATH="../libmetal/build/lib" `

Comment (tammyleino) [reply cmake/options.cmake]: @arnopo I disabled docs generation.  I don't know where all these "file renamed without changes" commits are coming from.  I didn't rename any files (and they aren't renamed) and I only staged the one cmake file for the latest commit.

File: doc/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+install (DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/html
+  DESTINATION share/doc/${PROJECT_NAME})
+install (FILES ${PROJECT_SOURCE_DIR}/README.md
+  DESTINATION share/doc/${PROJECT_NAME})
+install (FILES ${PROJECT_SOURCE_DIR}/LICENSE.md
+  DESTINATION share/doc/${PROJECT_NAME})
+
+endif (DOXYGEN_FOUND)
+
+# vim: expandtab:ts=2:sw=2:smartindent
```
Comment (arnopo) [inline doc/CMakeLists.txt:19]: Clean this inappropriate line

**Review by arnopo — APPROVED** [review]
just a line to clean then LGTM

File: doc/CMakeLists.txt (hunk 3 above)
Comment (tammyleino) [reply doc/CMakeLists.txt]: @arnopo  Done.

**Review by edmooring — APPROVED** [review]
Good for now. Maybe create an issue to clean up the warnings?

**Comment by arnopo** [comment]
> > Good for now. Maybe create an issue to clean up the warnings?
> 
> Done.

https://github.com/OpenAMP/open-amp/issues/458

**Comment by arnopo** [comment]
seems that there is merge conflict 
@tammyleino : Please rebase your PR on top of the main branch to fix the conflict

**Comment by arnopo** [comment]
@tammyleino : as discussed i create another PR to fix merge conflict and merge it:
https://github.com/OpenAMP/open-amp/pull/459

**Pull request closed without merging**